jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassURLHelperImpl.java

```diff
@@ -126,12 +125,14 @@ public boolean setMinimizeUpdateChecks() {
 		SharedClassFilter theFilter = getSharingFilter();
 		boolean doFind, doStore;
 		if (theFilter!=null) {
-			doFind = theFilter.acceptFind(className);
-			/* Don't invoke the store filter if the cache is full */
-			if (nativeFlags[CACHE_FULL_FLAG] == 0) {
-				doStore = theFilter.acceptStore(className);
-			} else {
-				doStore = true;
+			synchronized(theFilter) {
```

Comment:
Better to keep the old behavior and synchronize on the instance of SharedClassURLHelperImpl than to start synchronizing on the user supplied object, which could have unexpected side effects. Or best is to synchronize on an internal Object.

Reply:
Note the same synchronization changes are also needed in SharedClassURLClasspathHelperImpl and SharedClassTokenHelperImpl.

Reply:
I didn't find proper internal obj to sync in SharedClassURLHelperImpl and SharedClassTokenHelperImpl, so I am synchronizing on `this`.